lib/karafka/routing/route.rb

```diff
@@ -0,0 +1,73 @@
+module Karafka
+  module Routing
+    # Class representing a single route (from topic to worker) with all additional features
+    # and elements. Single route contains descriptions of:
+    # - topic - Kafka topic name (required)
+    # - controller - Class of a controller that will handle messages from a given topic (required)
+    # - group - Kafka group that we want to use (optional)
+    # - worker - Which worker should handle the backend task (optional)
+    # - parser - What parsed do we want to use to unparse the data (optional)
+    # - interchanger - What interchanger to encode/decode data do we want to use (optional)
+    class Route
+      # Only ASCII alphanumeric characters and underscore and dash are allowed in topics and groups
+      NAME_FORMAT = /\A(\w|\-)+\z/
+
+      # Options that we can set per each route
+      attr_writer :group, :topic, :controller, :worker, :parser, :interchanger
+
+      # This we can get "directly" because it does not have any details, etc
+      attr_reader :controller
```

Comment:
Maybe it is better to have ` attr_accessor :controller` and get rid of `:controller` within attr_writers?